### PR TITLE
Add test workflow

### PR DESCRIPTION
**Pull request by Dzejkop**
## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

**Review by iamrecursion — APPROVED** [review]
Looks fine, but how did we let the tests get lost like this?

File: tests/common/chain_mock.rs [hunk 1]
```diff
@@ -126,31 +130,24 @@ pub async fn spawn_mock_chain(
...
-    let (insert_verifiers, update_verifiers, identity_manager_impl) =
-        tokio::join!(insert_verifiers, update_verifiers, identity_manager_impl);
-
-    let insert_verifiers = insert_verifiers?;
-    let update_verifiers = update_verifiers?;
-    let identity_manager_impl = identity_manager_impl?;
+        .send()
+        .await?;
 
-    // TODO: This is sequential but could be parallelized.
```
Comment (gswirski) [inline tests/common/chain_mock.rs:151]: this TODO should stay? The new impl looks very sequential?

**Review by gswirski — APPROVED** [review]
nit: bring back TODO comment?

File: tests/common/chain_mock.rs (hunk 1 above)
Comment (Dzejkop) [reply tests/common/chain_mock.rs:151]: Yes, but that was the issue. Parallelizing it caused nonce issues. It's safer to just run everything sequentially until we have a good idea on how to handle that

**Pull request merged**